### DISCUSSION TITLE
Alternate (Failover) Destinations??

### DISCUSSION THREAD

**Discussion by GWhiteGE** [nextgenhealthcare/connect#6353] [Q&A] [answered]
I had a client ask me today if there is a way to reroute an HL7 result message out of Mirth to a different IP/Port destination if (and only if) the original destination was unreachable, essentially a redundant/failover server to receive HL7 results. I can't think of any way to do this but thought I would throw the question out to the community to get any suggestions, or confirmation that it is not doable. Thanks in advance.

Mirth v4.5.1 running on Windows 2019 Server, HL7 v2.x messaging

**Comment by kirbykn2** [comment, marked as answer]
Yep,

   - Destination 1 -  process all messages
   - Destination 2 - failover destination that processes only messages that
   failed at destination 1

if the message is successful stop processing, if not continue processing to
the second destination.

On Wed, Nov 13, 2024 at 12:01 PM GWhiteGE ***@***.***> wrote:

> I had a client ask me today if there is a way to reroute an HL7 result
> message out of Mirth to a different IP/Port destination if (and only if)
> the original destination was unreachable, essentially a redundant/failover
> server to receive HL7 results. I can't think of any way to do this but
> thought I would throw the question out to the community to get any
> suggestions, or confirmation that it is not doable. Thanks in advance.
>
> Mirth v4.5.1 running on Windows 2019 Server, HL7 v2.x messaging
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/nextgenhealthcare/connect/discussions/6353>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/APRXWD4MXVLPCTI5OBNRBIL2AOAXRAVCNFSM6AAAAABRW62OMOVHI2DSMVQWIX3LMV43ERDJONRXK43TNFXW4OZXGQ3TQMBTGU>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Best,

Kirby Knight
| 231.735.4650 | ***@***.***

**Reply by pacmano1** [reply]
What @kirbykn2 said.  

You can check the response status of the first destination in the second destination's filter and `return false` in that filter if the message was sent.

**Reply by GWhiteGE** [reply]
Thank you both for the tip. To be sure I am on the right track now, I think what I need to do is to create a Response Transformer in the first Destination connector that captures responseStatus variable and makes it available in the ChannelMap

```
_var mapping;
try {
    mapping = '';
} catch (e) {
    mapping = '';
}
channelMap.put('responseStatus', validate(mapping, '', new Array()));_
```

and then use the value of that variable in a Filter in second Destination (Failover) connector.

```
_if ($('responseStatus') != SENT) {
    return true;
}
return false;_
```

**Reply by pacmano1** [reply]
The response status for destination 1 is available to reference in the filter for destination 2, you don't need a new variable for it.

**Reply by pacmano1** [reply]
Assuming your primary destination number ID is 1, in the filter of the second destination:

```JS
return $('d1').getStatus() != 'SENT'
```

In other words, you don't need any of your code.

**Reply by GWhiteGE** [reply]
Got it! Thanks again for the help!!
